---
title: "BETA_integration_PAX8"
output: html_document
---

```{r setup, include=FALSE}
library(tidyverse)
```

# Input data

- PAX8 DEG obtained for cell line 786M1A by two different methods (edgeR and DESeq2).
- PAX8 narrow peaks obtained for the cell line 7861MA by macs2.

# BETA integration

BETA can integrate RNA-seq and ChIP-seq data to obtain a list of target genes.
We performed BETA using DEG obtained considering constructs together and also considering constructs separated.

Objective: Compare target genes in both constructs and detected by both methods (edgeR and DESeq2)

## 1. RNA-seq analysis considering different constructs as replicates (constructs together)

Threshold:

d 50000
c 0.001
df 0.05
da 1

```{r}
# obtain list target genes when considered both constructs together
PAX8_target_7861MA_DESeq2_ct <-read.table('/shares/MRC-CU/Samarajiwa/Natalia/Integration_PAX8/BETA/constructs_together/output_DESEq2_df0.05_d50_c0.001/PAX8_DESeq2_df0.05_d50_c0.001_downtarget.txt', header = TRUE, sep = '\t')

PAX8_target_7861MA_edgeR_ct <-read.table('/shares/MRC-CU/Samarajiwa/Natalia/Integration_PAX8/BETA/constructs_together/output_edgeRLRT_df0.05_d50_c0.001/PAX8_edgeRLRT_df0.05_d50_c0.001_downtarget.txt', header = TRUE, sep = '\t')

nrow(PAX8_target_7861MA_DESeq2_ct)
nrow(PAX8_target_7861MA_edgeR_ct)
```

```{r}
# obtain intersect between them
PAX8_target_7861MA_ct <- inner_join(PAX8_target_7861MA_DESeq2_ct, PAX8_target_7861MA_edgeR_ct, by = "GeneSymbol")
nrow(PAX8_target_7861MA_ct)
```

Results: good correlation between both methods edgeR and DESeq2.

## 2. RNA-seq analysis considering different constructs independently (constructs separated)

Threashold DESeq2 and edgeR 786:

d 50000
c 0.001
df 0.05
da 1

Threashold DESeq2 1503:

d 50000
c 0.2
df 0.05
da 1

Threashold edgeR 1503:

d 50000
c 0.1
df 0.05
da 1


```{r}
# obtain list target genes when considered both constructs separated
PAX8_target_7861MA_DESeq2_786 <-read.table('/shares/MRC-CU/Samarajiwa/Natalia/Integration_PAX8/BETA/constructs_separated/output_DESeq2_786_df0.05_d50_c0.001/PAX8_DESeq2_786_df0.05_d50_c0.001_downtarget.txt', header = TRUE, sep = '\t')
nrow(PAX8_target_7861MA_DESeq2_786)

PAX8_target_7861MA_DESeq2_1503 <-read.table('/shares/MRC-CU/Samarajiwa/Natalia/Integration_PAX8/BETA/constructs_separated/output_DESeq2_1503_df0.05_d50_c0.2/PAX8_DESeq2_1503_df0.05_d50_c0.2_downtarget.txt', header = TRUE, sep = '\t')
nrow(PAX8_target_7861MA_DESeq2_1503)

PAX8_target_7861MA_edgeRLRT_786 <-read.table('/shares/MRC-CU/Samarajiwa/Natalia/Integration_PAX8/BETA/constructs_separated/output_edgeRLRT_786_df0.05_d50_c0.001/PAX8_edgeRLRT_786_df0.05_d50_c0.001_downtarget.txt', header = TRUE, sep = '\t')
nrow(PAX8_target_7861MA_edgeRLRT_786)

PAX8_target_7861MA_edgeRLRT_1503 <-read.table('/shares/MRC-CU/Samarajiwa/Natalia/Integration_PAX8/BETA/constructs_separated/output_edgeRLRT_1503_df0.05_d50_c0.1/PAX8_edgeRLRT_1503_df0.05_d50_c0.1_downtarget.txt', header = TRUE, sep = '\t')
nrow(PAX8_target_7861MA_edgeRLRT_1503)
```

```{r}
# intersection between different tools (edgeR and DESeq2)
PAX8_target_7861MA_786 <- inner_join(PAX8_target_7861MA_DESeq2_786, PAX8_target_7861MA_edgeRLRT_786, by = "GeneSymbol")
nrow(PAX8_target_7861MA_786)

PAX8_target_7861MA_1503 <- inner_join(PAX8_target_7861MA_DESeq2_1503, PAX8_target_7861MA_edgeRLRT_1503, by = "GeneSymbol")
nrow(PAX8_target_7861MA_1503)

# intersection between different constructs
PAX8_target_7861MA_cs <- inner_join(PAX8_target_7861MA_786, PAX8_target_7861MA_1503, by = "GeneSymbol")
nrow(PAX8_target_7861MA_cs)
```

Results: good correlation between both methods edgeR and DESeq2.
We lose many results that were detected before when combining both constructs. 

```{r}
# intersection between targets when considering constructs together vs separated
PAX8_target_7861MA_only_ct <- anti_join(PAX8_target_7861MA_ct, PAX8_target_7861MA_cs, by = "GeneSymbol")
nrow(PAX8_target_7861MA_only_ct)
PAX8_target_7861MA_only_cs <- anti_join(PAX8_target_7861MA_cs, PAX8_target_7861MA_ct, by = "GeneSymbol")
nrow(PAX8_target_7861MA_only_cs)
```


We can see that all targets that were detected when analysing data coming from constructs separated (435) were detected also in constructs together (252), so we're not missing any information but we have other target genes that are not detected individually.


```{r}
# write results
write_csv(PAX8_target_7861MA_cs, '/shares/MRC-CU/Samarajiwa/Natalia/Integration_PAX8/BETA/targets/PAX8_target_7861MA_cs.csv')
write_csv(PAX8_target_7861MA_ct, '/shares/MRC-CU/Samarajiwa/Natalia/Integration_PAX8/BETA/targets/PAX8_target_7861MA_ct.csv')
```

I keep the target genes detected by both constructs independently and by both methods (more stringent)

We obtain only the genes regulated at the promoter level: distance +- 10000 TSS

Because we are only considering common genes detected by both constructs and both tools, the distance info can be obtained from each of the four files, we use DESeq2 and construct 786, but could be any file. 

```{r}
# obtain file with distance information from the TSS
peaks_distance_info <- read.table('/shares/MRC-CU/Samarajiwa/Natalia/Integration_PAX8/BETA/constructs_separated/output_DESeq2_786_df0.05_d50_c0.001/PAX8_DESeq2_786_df0.05_d50_c0.001_downtarget_associate_peaks.bed', header = TRUE)
# select genes with peaks around +-10000 TSS
genes_peak_prom <- peaks_distance_info %>% filter(Distance >= -10000 & Distance <= 10000) %>% rename(GeneSymbol = "Symbol")

# obtain PAX8 target genes in the promoter
PAX8_target_7861MA_prom <- inner_join(PAX8_target_7861MA_cs, genes_peak_prom, by = "GeneSymbol")
nrow(PAX8_target_7861MA_prom)
write_csv(PAX8_target_7861MA_prom, '/shares/MRC-CU/Samarajiwa/Natalia/Integration_PAX8/BETA/targets/PAX8_target_7861MA_prom10000.csv')
```

```{r}
# obtain list of targets (just symbol and ensembl ID)
PAX8_target_7861MA_prom_list <- PAX8_target_7861MA_prom %>% select(GeneSymbol, refseqID.x.x) %>%
  unique()
nrow(PAX8_target_7861MA_prom_list)
write_csv(PAX8_target_7861MA_prom_list, '/shares/MRC-CU/Samarajiwa/Natalia/Integration_PAX8/BETA/targets/PAX8_target_list_7861MA_prom10000.csv')
```

